store/store.go

```diff
@@ -2372,30 +2373,46 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 	return snaps, nil
 }
 
+// abbreviated info structs just for the download info
+type storeInfoChannelAbbrev struct {
+	Download storeSnapDownload `json:"download"`
+}
+
+type storeInfoAbbrev struct {
+	// discard anything beyond the first entry
+	ChannelMap [1]storeInfoChannelAbbrev `json:"channel-map"`
+}
+
 var errUnexpectedConnCheckResponse = errors.New("unexpected response during connection check")
 
 func (s *Store) snapConnCheck() ([]string, error) {
 	var hosts []string
-	// TODO: move this to SnapAction:download when that's done (and maybe
-	//       avoid doRequest with its compulsory macaroon refresh dance)
 	// NOTE: "core" is possibly the only snap that's sure to be in all stores
 	//       when we drop "core" in the move to snapd/core18/etc, change this
-	deetsURL := s.endpointURL(path.Join(detailsEndpPath, "core"), url.Values{"fields": {"anon_download_url"}})
-	hosts = append(hosts, deetsURL.Host)
-
-	var remote snapDetails
-	resp, err := httputil.RetryRequest(deetsURL.String(), func() (*http.Response, error) {
-		return s.doRequest(context.TODO(), s.client, &requestOptions{Method: "GET", URL: deetsURL}, nil)
+	infoURL := s.endpointURL(path.Join(snapInfoEndpPath, "core"), url.Values{
+		"fields":       {"download"},
+		"architecture": {"amd64"},
```

Comment:
Curious, why hardcode the architecture here?

Reply:
I just need to specify one. I could also use the store's, I guess :-)

Reply:
Can you add a note explaining this.